Models/AbstractProjectModels/Validation/Data/Admin/Head/AbstractValidator.php

```diff
@@ -0,0 +1,113 @@
+<?php
+declare(strict_types=1);
+
+namespace Models\AbstractProjectModels\Validation\Data\Admin\Head;
+
+use http\Exception\InvalidArgumentException;
+use Interfaces\Admin\AdminDataValidatorInterface;
+
+abstract class AbstractValidator implements AdminDataValidatorInterface
+{
+    private const FIELD_NAMES = [];
+    private const FIELD_VALUES = [];
+    protected array $fieldNames = [];
+    protected array $fieldValues = [];
+    protected string $fieldName;
+    protected string $fieldValue;
+
+    public function emptyCheckBox(array $data): array
+    {
+        if (empty($data)) {
+            $data['empty_data'] = false;
+        }
+
+        return $data;
+    }
+
+    public function correctCheckBox(array $data, string $actionName = null): array
+    {
+        $result = [];
+        foreach ($data as $adminLogin => $adminData) {
+            $result[$adminLogin] = $this->validateCheckBoxData($this->splitString($adminData), $actionName);
+        }
+
+        return $result;
+//        return $this->validateCheckBoxData($this->splitString($data['change']), $actionName);
+    }
+
+    protected function splitString(string $string): ?array
+    {
+        $result = [];
```

Comment:
i looked and found that it can be like this maybe?
$result = [];
        $fields = [
            'id_value',
            'field_name',
            'field_value'
        ];
        $splits = explode('/', trim($string, '/'));
        if (!empty($splits)) {
            $i = 0;
            foreach ($fields as $field) {
                if (isset($splits[$i])) {
                    if (is_numeric($splits[$i])) {
                        $result[$field] = (int)$splits[$i];
                    } else {
                        $result[$field] = $splits[$i];
                    }
                } else {
                    $result[$field] = '';
                }
            }
            $i++;
        }
        If your idea better, I would be pleased to look at that. 
        xo xo

Reply:
i dont know why but my empty spaces deleted in this code, sorry for bad look